CIScripts/Linux/BuildLinuxAgent.sh

```diff
@@ -0,0 +1,42 @@
+# Launches a Vagrant VM and builds vRouter Agent.
+# Run this script from the directory that contains Vagrantfile.
+
+if [ "$#" -ne 5 ]; then
+    echo "Usage: TOOLS_BRANCH CONTROLLER_BRANCH VROUTER_BRANCH GENERATEDS_BRANCH SANDESH_BRANCH"
+    exit 1
+fi
+
+set -e
+
+if [ ! -d ".vagrant" ]; then
+    echo " !!! WARNING:"
+    echo ".vagrant directory not found! This may cause cleanup issues!!!"
+    echo "see if any old virtualbox processes are running just in case:"
+    ps aux | grep virtualbox
+    echo "continuing..."
+fi
+
+vagrant halt
+vagrant destroy -f
+vagrant up
+vagrant ssh -c "
+ls
+cd contrail-vrouter
+pushd controller
+  git checkout $2
```

Comment:
```
ARGUMENT DISAMBIGUATION
When there is only one argument given and it is not -- (e.g. "git checkout abc"), and when the argument is both a valid <tree-ish> (e.g. a branch "abc" exists) and a valid <pathspec> (e.g. a file or a directory whose name is "abc" exists), Git would usually ask you to disambiguate. Because checking out a branch is so common an operation, however, "git checkout abc" takes "abc" as a <tree-ish> in such a situation. Use git checkout -- <pathspec> if you want to checkout these paths out of the index.
```
https://git-scm.com/docs/git-checkout
What's wrong with how it is right now?